doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+
+A problem arises here in that encouraging sparse solutions doesn't necessarily generate interpretable counterfactuals. This happens because the loss doesn't prevent the counterfactual solution from moving off the data distribution. Thus, you will likely get an answer that doesn't look like something that you'd expect to see from the data.
+
+__TODO:__
+- Picture example. Something similar to: https://github.com/interpretml/DiCE
+
+**Pros**
+- Both a black and white box method
+- Doesn't require access to the training dataset
+
+**Cons**
+- Not likely to give human interpretable instances
+- Requires configuration in the choice of $\lambda$
+
+
+#### Contrastive Explanation Method
+
+| Model-types | Task-types     | Data-types  |
+| ----------- | -------------- | ----------- |
+| TF/Kera     | Classification | Tabular     |
+| Black-box   |                | Image       |
+
+
+CEM follows a similar approach to the above but includes three new details. Firstly an elastic net $\beta L_{1} + L_{2}$ regularizer term is added to the loss. This term causes the solutions to be both close to the original instance and sparse.
+
+Secondly, we require that $\delta$ only adds new features rather than takes them away. We need to define what it means for a feature to be present so that the perturbation only works to add and not remove them. In the case of the MNIST dataset, an obvious choice of "present" feature is if the pixel is equal to 1 and absent if it is equal to 0. This is simple in the case of the MNIST data set but more difficult in complex domains such as color images.
+
+Thirdly, by training an optional autoencoder to penalize counter factual instances that deviate from the data distribution. This works by minimizing the reconstruction loss of the autoencoder applied to instances. If a generated instance is unlike anything in the dataset, the autoencoder will struggle to recreate it well, and its loss term will be high. We require two hyperparameters $\beta$ and $\gamma$ to define the following Loss,
+
+$$L(X'|X)= (f_{t}(X') - p_{t})^2 + \beta L_{1}(X', X) + L_{2}(X', X)^2 + \gamma L_{2} (X', AE(X'))^2$$
+
+This approach extends the definition of interpretable to include a requirement that the computed counterfactual be believably a member of the dataset. It turns out that minimizing this loss isn't enough to always get interpretable results. And in particular, the constructed counterfactual often doesn't look like a member of the target class.
+
+Similar to the previous method, this method can apply to both black and white box models. In the black-box case, there is still a performance cost from computing the numerical gradients.
+
+__TODO:__
+- Picture example of results including less interpretable ones.
+
+**Pros**
+- Provides more interpretable instances than the counterfactual instances' method.
+
+**Cons**
+- Requires access to the dataset to train the autoencoder
```

Comment:
I would also separate out that the autoencoder needs to be pre-trained (so needs to be defined first of all, then trained).